components/javascript-wrapper.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * This file contacts the common code used by all of the variations of any RDF
+ * node componentin the noflo implementation of the RDF Pipeline.
+ */
+
+var _ = require('underscore');
+
+// TODO: Revisit with James' latest work
+var promiseComponent = require('./promise-component');
+var rpf = require('./rpf');
+
+/**
+ * This module provides a Javascript wrapper for updaters.  It is responsible for setting up the
+ * code to ensure updater callback, and then invoking the RDF pipeline framework to complete building
+ * the component and setting up the vni and state metadata.
+ *
+ * @param nodeDef updater node definition. This can be a simple updater function, or an object that contains 
+ *        updater metadata (e.g., input/output port definitions, description, icons), and an updater.
+ *
+ * @return a promise to create the noflo rdf component
+ */
+module.exports = function(nodeDef) { 
+
+    // Save the caller's node instance 
```

Comment:
Is this comment correct?  I would think that the nodeDef parameter would be the definition of the node that we wish to create, so I am confused about what node instance this would be talking about.  I am confused because if nodeDef is being passed in, then presumably the node instance has not yet been created.